libs/utils/src/lib/location-ext.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import { Router, NavigationEnd, NavigationExtras } from '@angular/router';
+import { filter, skip, take } from 'rxjs/operators';
+import { Location } from '@angular/common';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LocationExtService implements OnDestroy {
+  private _isFirstRoute = true;
+
+  additionalNavigationSubscription = this.router.events
+    .pipe(
+      filter(evt => {
+        return evt instanceof NavigationEnd;
+      }),
+      skip(1),
+      take(1)
+    )
+    .subscribe(() => {
+      this._isFirstRoute = false;
+    });
+
+  constructor(
+    private readonly router: Router,
+    private readonly location: Location
+  ) {}
+
+  ngOnDestroy() {
+    this.additionalNavigationSubscription.unsubscribe();
+  }
+
+  isInitialRoute() {
```

Comment:
This and additionalNavigationSubscription are only used in tests, right?

Should they be removed?